File: backend/src/main/java/com/github/jadamon42/family/model/Relationship.java
package com.github.jadamon42.family.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Value;
import lombok.With;

import java.util.Set;
import java.util.UUID;

@Value
@With
@Builder
@AllArgsConstructor
public class Relationship {
    Set<UUID> pathIds;
    String relationshipLabel;
    String inverseRelationshipLabel;
}

File: frontend/src/renderer/models/Relationship.ts
class Relationship {
  pathIds: string[];
  relationshipLabel: string;
  inverseRelationshipLabel: string;

  constructor(
    pathIds: string[],
    relationshipLabel: string,
    inverseRelationshipLabel: string
  ) {
    this.pathIds = pathIds;
    this.relationshipLabel = relationshipLabel;
    this.inverseRelationshipLabel = inverseRelationshipLabel;
  }
}

export default Relationship;
